### PR TITLE
Add methods to read and write data

### PR DESCRIPTION
**Pull request by SergSkol**
-Use JSON module
-Add methods to save and load data
-Add id to class book

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -95,4 +97,106 @@ def list_rentals
...
+
+  def load_data
+    @books = load_books || []
+    @people = load_people || []
+    @rentals = load_rentals || []
+  end
+
+  def save_books
+    # File.write('./data/books.json', JSON.generate(@books))
+    # File.write('./data/books.json', Marshal.dump(@books))
```
Comment (GzimAsani) [inline app.rb]: - [ ] Please remove the commented out code, it makes your code looks unprofessional.

File: app.rb [hunk 2]
```diff
@@ -95,4 +97,106 @@ def list_rentals
...
+      id = e['id']
+      title = e['title']
+      author = e['author']
+      Book.new(title, author, id)
+    end
+  end
+
+  def save_people
+    # File.write('./data/people.json', JSON.generate(@people))
+    # File.write('./data/people.json', Marshal.dump(@people))
```
Comment (GzimAsani) [inline app.rb]: - [ ] Please remove the commented out code, it makes your code looks unprofessional.

File: app.rb [hunk 3]
```diff
@@ -95,4 +97,106 @@ def list_rentals
...
+    arr = []
+    @people.each do |e|
+      arr.push(e.receive_item)
+    end
+    File.write('./data/people.json', JSON.generate(arr))
+  end
+
+  def load_people
+    # JSON.parse(File.read('./data/people.json')) if File.exist?('./data/people.json')
+    # Marshal.load(File.read('./data/people.json')) if File.exist?('./data/people.json')
```
Comment (GzimAsani) [inline app.rb]: - [ ] Please remove the commented out code, it makes your code looks unprofessional.

File: app.rb [hunk 4]
```diff
@@ -95,4 +97,106 @@ def list_rentals
...
+      else
+        specialization = e['specialization']
+        Teacher.new(age, name, specialization, id)
+      end
+    end
+  end
+
+  def save_rentals
+    # File.write('./data/rentals.json', JSON.generate(@rentals))
+    # File.write('./data/rentals.json', Marshal.dump(@rentals))
```
Comment (GzimAsani) [inline app.rb]: - [ ] Please remove the commented out code, it makes your code looks unprofessional.

File: app.rb [hunk 5]
```diff
@@ -95,4 +97,106 @@ def list_rentals
...
+    arr = []
+    @rentals.each do |e|
+      arr.push(e.receive_item)
+    end
+    File.write('./data/rentals.json', JSON.generate(arr))
+  end
+
+  def load_rentals
+    # JSON.parse(File.read('./data/rentals.json')) if File.exist?('./data/rentals.json')
+    # Marshal.load(File.read('./data/rentals.json')) if File.exist?('./data/rentals.json')
```
Comment (GzimAsani) [inline app.rb]: - [ ] Please remove the commented out code, it makes your code looks unprofessional.

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @SergSkol and @studiosnabeel             :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with data preservation, it works as expected   :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 
 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (SergSkol) [reply app.rb]: Done, thanks!

File: app.rb (hunk 2 above)
Comment (SergSkol) [reply app.rb]: Done, thanks!

File: app.rb (hunk 4 above)
Comment (SergSkol) [reply app.rb]: Done, thanks!

**Review by danmainah — APPROVED** [review]
Hi @SergSkol , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights

✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**